repository/src/main/java/org/alfresco/repo/domain/permissions/AclDAO.java

```diff
@@ -182,4 +182,9 @@
      * @return Long
      */
     public Long getMaxChangeSetIdByCommitTime(long maxCommitTime);
+
+    /**
+     * @return the current AclChangeSet created for the current ACL change 

```

Comment:
This javadoc is not clear. "Current" means for current transaction, if it is present. The return value can be null.

The AclChangeSet is never used in this interface, do you know why?

Reply:
I can change to: "@return the current AclChangeSet created for the current transaction. It can be null if a transaction is not active."
AclChangeSet is used only inside the implementation. In the interface it uses only field of this object, for instance in getMaxChangeSetIdByCommitTime and getMaxChangeSetCommitTime.

Reply:
AclChangeSet is a DAO layer entity and should not be exposed via the interface :)

Reply:
I apologize because they are in the same package the layering wasn't clear to me. Anyway I did a refactoring in order to have a method like  getMaxChangeSetIdByCommitTime